common/optimiser/Optimiser.java

```diff
@@ -56,12 +57,16 @@ public Optimiser() {
     }
 
     public synchronized Status optimise(long timeLimitMillis) {
+        if (hasSolver && constraintsChanged()) {
+            constraints.forEach(constraint -> constraint.initialise(solver));
+            status = Status.NOT_SOLVED;
+        }
```

Comment:
New set of constraints depends on knowing the traversal costs, so the constraints may change along with the objective function coefficients.